### ISSUE TITLE
[Bug]: Catalog filtered table lingers in table runtime list

### ISSUE THREAD

**Issue by link3280** [apache/amoro#2642]
### What happened?

Catalog allows filtering db and tables through `database.filter-regular-expression` and `table-filter`. However, it's not effective for the able runtime list, so they still show up in the table list. If users click the link and access the table detail, there would a 404 error.

### Affects Versions

0.5.1

### What engines are you seeing the problem on?

AMS

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by azhsmesos** [comment]
I hope I can try to solve this issue, but I have not known amoro before, it may take a little time, can I have a try?

**Comment by zhoujinsong** [comment]
> I hope I can try to solve this issue, but I have not known amoro before, it may take a little time, can I have a try?

Of course, if you need any assistance during the work, please let me know.

**Comment by zhoujinsong** [comment]
It seems this bug no longer exists in the master branch. Can you help to check again? @link3280

**Comment by link3280** [comment]
Sure. Let's revisit the issue in 0.7-rc.
